engine/src/WindowGLFW.cpp

```diff
@@ -0,0 +1,52 @@
+#include "Engine/Core/Rendering/Window/WindowGLFW.hpp"
+
+#include "Engine/Core/Debug/Log.hpp"
+#include "Engine/Core/Tools/Format.hpp"
+
+using namespace Engine::Core::Renderering;
+using namespace Engine::Core::Tools;
+using namespace Engine::Core::Debug;
+using namespace std;
+
+void WindowGLFW::initialize(const WindowCreateArg &arg) noexcept
+{
+    Log::logInitializationStart("GLFW window");
+
+    // Init glfw
+    if (!glfwInit())
+    {
+        FUNCT_ERROR("glfwInit failed");
+        exit(EXIT_FAILURE);
+    }
+
+#ifdef _DEBUG
+    glfwSetErrorCallback([](int errorCode, const char *description) {
+        Log::logError(stringFormat("GLFW error (%x) = %s\n", errorCode, description));
+    });
+#endif
+
+    glfwWindowHint(GLFW_CONTEXT_VERSION_MAJOR, 4);
+    glfwWindowHint(GLFW_CONTEXT_VERSION_MINOR, 6);
+    glfwWindowHint(GLFW_OPENGL_DEBUG_CONTEXT, GLFW_TRUE);
+    glfwWindowHint(GLFW_OPENGL_PROFILE, GLFW_OPENGL_CORE_PROFILE);
+
+    m_window = glfwCreateWindow(arg.width, arg.height, arg.name.c_str(), nullptr, nullptr);
+
+    if (!m_window)
+    {
+        FUNCT_ERROR("glfwCreateWindow failed");
```

Comment:
On my machine, as is, the build crashes here:
![image](https://user-images.githubusercontent.com/33126700/106213883-90c1be80-61c5-11eb-8bbd-2562a1ba07d3.png)

Adding this into `main.cpp` solved the problem:
```
extern "C"
{
    _declspec(dllexport) int NvOptimusEnablement = 1;
    _declspec(dllexport) int AmdPowerXpressRequestHighPerformance = 1;
}
```

Reply:
Ahah that happened XD